pulp_2to3_migration/tests/functional/test_multiple_plugins.py

```diff
@@ -0,0 +1,145 @@
+        cls.file_repo_versions_api = RepositoriesFileVersionsApi(file_client)
+        cls.file_remote_api = RemotesFileApi(file_client)
+        cls.file_distribution_api = DistributionsFileApi(file_client)
+        cls.file_publication_api = PublicationsFileApi(file_client)
+        cls.file_content_api = ContentFilesApi(file_client)
+
+        # Create api clients for RPM
+        cls.rpm_repo_api = RepositoriesRpmApi(rpm_client)
+        cls.rpm_repo_versions_api = RepositoriesRpmVersionsApi(rpm_client)
+        cls.rpm_remote_api = RemotesRpmApi(rpm_client)
+        cls.rpm_distribution_api = DistributionsRpmApi(rpm_client)
+        cls.rpm_publication_api = PublicationsRpmApi(rpm_client)
+        cls.rpm_content_apis = {
+            'advisory': ContentAdvisoriesApi(rpm_client),
+            'disttree': ContentDistributionTreesApi(rpm_client),
+            'modulemd': ContentModulemdsApi(rpm_client),
+            'modulemd-defaults': ContentModulemdDefaultsApi(rpm_client),
+            'category': ContentPackagecategoriesApi(rpm_client),
+            'environment': ContentPackageenvironmentsApi(rpm_client),
+            'group': ContentPackagegroupsApi(rpm_client),
+            'langpack': ContentPackagelangpacksApi(rpm_client),
+            'package': ContentPackagesApi(rpm_client),
+        }
+
+        # Create api clients for Migration
+        cls.migration_plans_api = MigrationPlansApi(migration_client)
+        cls.pulp2content_api = Pulp2ContentApi(migration_client)
+        cls.pulp2repositories_api = Pulp2RepositoriesApi(migration_client)
+
+    @classmethod
+    def run_migration(cls, plan, run_params={}):
+        """
+        Run a migration plan.
+
+        Args:
+            plan(str): A migration plan to run, in JSON format.
+            run_params(dict): parameters for the `run` call. Optional.
+
+        Returns:
+            task(pulpcore.app.models.Task): a migration task created for this plan
+
+        """
+        mp = cls.migration_plans_api.create({'plan': plan})
+        mp_run_response = cls.migration_plans_api.run(mp.pulp_href, run_params)
+        task = monitor_task(mp_run_response.task)
+        monitor_task_group(task.task_group)
+        return task
+
+    @classmethod
+    def tearDownClass(cls):
+        """
+        Clean up the database after the set of tests is run.
+        """
+        cmd = get_psql_smash_cmd(TRUNCATE_TABLES_QUERY_BASH)
+        cls.smash_cli_client.run(cmd, sudo=True)
+        time.sleep(0.5)
+
+
+class TestRpmIsoMigration(BaseTestMultiplePlugins, unittest.TestCase):
+    """
+    Test RPM and ISO migration
+    """
+    def test_rpm_iso_migration_sequential(self):
+        """Test migrating RPM plugin and ISO plugin in two separate runs"""
+        set_pulp2_snapshot(name='rpm_base_4repos')
+        self.run_migration(RPM_SIMPLE_PLAN)
+
+        set_pulp2_snapshot(name='file_base_4repos')
+        self.run_migration(FILE_SIMPLE_PLAN)
+
+        rpm_repo_count = 3
```

Comment:
took me a while to understand why the count is 3. in that snapshot are 5 repos, from which 2 are empty. not sure what to suggest to make this more explicit?..and why this is called rpm_base_4repos while in the snapshot there are 5? https://github.com/pulp/pulp-2to3-migration-test-fixtures/blob/master/rpm_base_4repos/description.md

Reply:
It's very confusing I agree. 
I noticed the snapshot naming while prepping this pr as well. I think originally it was 4 and then we tweaked it and never changed the name. I'll try to redo it at some point to avoid confusion around 4 in name vs 5 in reality.
Simple plan does not migrate empty repos, so I can provide complex plan or leave a comment that empty repos are not migrated using simple plan.